### PR TITLE
Improve Configuration Management

### LINKED ISSUE #22: On SDK 11 & 12 blank SoftDevice is impossible
For SDK11 and 12 softDevice is forced by the makefile.

### PR DESCRIPTION
**Pull request by fletort**
- Guess NRF_MODEL from NRF_IC is set
 - Force NRF_MODEL for old SDK (9, 10)
 - Default SDK depending NRF_IC
 - Default NRF_IC depending NRF_MODEL
 - Default RAM/FLASH depending NRF_IC
 - Bug: No more force SOFTDEVICE on SDK 11 and 12. (black soft device can not be used with these SDK). Close #22.

### CODE REVIEW

File: make/Makefile [hunk 1]
```diff
@@ -41,107 +42,152 @@ OBJDUMP = $(TOOLCHAIN_PATH)$(TOOLCHAIN_PREFIX)-objdump
...
+	
+	# SDK 11
+	ifeq ($(SDK_VERSION), 11)
+		ifeq ($(SOFTDEVICE_MODEL), s130)
+			USE_BLE = 1
+			WAITED_SOFTDEVICE_VERSION ?= 2.0.0
+			WAITED_NRF_MODEL = nrf51
+		else ifeq ($(SOFTDEVICE_MODEL), s132)
+			USE_BLE = 1
+			SOFTDEVICE_VERSION ?= 2.0.0
```
Comment (brghena) [inline make/Makefile]: `WAITED_NRF_MODEL = nrf52`

File: make/Makefile [hunk 2]
```diff
@@ -41,107 +42,152 @@ OBJDUMP = $(TOOLCHAIN_PATH)$(TOOLCHAIN_PREFIX)-objdump
...
+	
+	# SDK 12
+	ifeq ($(SDK_VERSION), 12)
+		ifeq ($(SOFTDEVICE_MODEL), s130)
+			USE_BLE = 1
+			WAITED_SOFTDEVICE_VERSION ?= 2.0.1
+			WAITED_NRF_MODEL = nrf51
+		else ifeq ($(SOFTDEVICE_MODEL), s132)
+			USE_BLE = 1
+			SOFTDEVICE_VERSION ?= 3.0.0
```
Comment (brghena) [inline make/Makefile]: `WAITED_NRF_MODEL = nrf52`

File: make/Makefile [hunk 3]
```diff
@@ -41,107 +42,152 @@ OBJDUMP = $(TOOLCHAIN_PATH)$(TOOLCHAIN_PREFIX)-objdump
...
-# endif
 
+#------ Now select/check SOFTDEVICE version if used (depending SDK version)
+ifdef SOFTDEVICE_MODEL
+	# SDK 9 or 10
+	ifneq (,$(filter $(SDK_VERSION),9 10))
+		WAITED_NRF_MODEL = nrf51
+		ifeq ($(SOFTDEVICE_MODEL), s110)
+		    USE_BLE = 1
+		    WAITED_SOFTDEVICE_VERSION ?= 8.0.0
```
Comment (brghena) [inline make/Makefile]: I think something went wrong here.

In each of these cases, shouldn't both a SOFTDEVICE_VERSION and a WAITED_SOFTDEVICE_VERSION be set? (SOFTDEVICE_VERSION conditionally and WAITED_SOFTDEVICE_VERSION unconditionally)

**Comment by brghena** [comment]
Yup. Looks like it works just fine for our projects. I've also updated the blink application to add a `SOFTDEVICE_MODEL`

**Comment by brghena** [comment]
@fletort I was testing the SDK12 issue and managed to mistakenly merge this PR. Oops!

But I was pretty close to happy with it anyways, so I will make the changes I requested. If you've got any reason why the changes shouldn't be made though, do let me know.

Thanks for the PR!

File: make/Makefile (hunk 3 above)
Comment (fletort) [inline make/Makefile]: Yes. but as in each case The two value will be the same, i do it only one time after all the if..
Like that i have only one line : 

SOFTDEVICE_VERSION ?= WAITED_SOFTDEVICE_VERSION

Comment (fletort) [inline make/Makefile]: my line 181

**Pull request merged**